File: src/gestorAplicacion/personas/Persona.java
package gestorAplicacion.personas;

import java.time.LocalDate;
import java.util.ArrayList;

import gestorAplicacion.general.DiaSemana;
import gestorAplicacion.general.Inventario;
import gestorAplicacion.ventas.Servicio;

import java.io.Serializable;

/*
 * Esta clase funciona como la clase padre para las clases cliente y empleado, por tanto maneja los atributos basicos de una persona
 * 
 * @author Mateo Alvarez Lebrum
 * @author Alejandro Alvarez Botero
 * @author Miguel Angel Barrera Bustamante
 * @author Alejandra Barrientos Grisales
 *
 */
public class Persona implements Serializable {
	private static final long serialVersionUID = 1L;

	protected String nombre;
	protected String telefono;
	protected String email;
	protected String identificacion;
	protected TipoDocumento tipoDeIdentificacion;
	protected Sexo sexo;

	/**
	 * @param nombre
	 * @param telefono
	 * @param email
	 * @param identificacion
	 * @param tipoDeIdentificacion
	 * @param sexo
	 */
	public Persona(String nombre, String telefono, String email, String identificacion,
			TipoDocumento tipoDeIdentificacion, Sexo sexo) {
		super();
		this.nombre = nombre;
		this.telefono = telefono;
		this.email = email;
		this.identificacion = identificacion;
		this.tipoDeIdentificacion = tipoDeIdentificacion;
		this.sexo = sexo;
		Inventario.agregarPersona(this);
	}

	public String getNombre() {
		return nombre;
	}

	public void setNombre(String nombre) {
		this.nombre = nombre;
	}

	public String getTelefono() {
		return telefono;
	}

	public void setTelefono(String telefono) {
		this.telefono = telefono;
	}

	public String getEmail() {
		return email;
	}

	public void setEmail(String email) {
		this.email = email;
	}

	public String getIdentificacion() {
		return identificacion;
	}

	public void setIdentificacion(String identificacion) {
		this.identificacion = identificacion;
	}

	public TipoDocumento getTipoDeIdentificacion() {
		return tipoDeIdentificacion;
	}

	public void setTipoDeIdentificacion(TipoDocumento tipoDeIdentificacion) {
		this.tipoDeIdentificacion = tipoDeIdentificacion;
	}

	public Sexo getSexo() {
		return sexo;
	}

	public void setSexo(Sexo sexo) {
		this.sexo = sexo;
	}

	/**
	 * @param contrato
	 * @param cargo
	 * @param servicio
	 * @param diasLaborales
	 * 
	 * @return crea una instancia de empleado
	 */
	public void contratar(Contrato contrato, String cargo, Servicio servicio, ArrayList<DiaSemana> diasLaborales) {
		new Empleado(this, contrato, cargo, servicio, diasLaborales);
	}

	/**
	 * @return String con la informacion de la persona
	 */
	public String mostrarInformacion() {
		return "Soy " + nombre + " con numero de indentificacion: " + identificacion;
	}

}

File: python/src/gestorAplicacion/personas/Persona.py
from gestorAplicacion.general.DiaSemana import DiaSemana
from gestorAplicacion.personas.Contrato import Contrato
from gestorAplicacion.ventas.Servicio  import Servicio

class Persona:

    def __init__(self, nombre, telefono, email, identificacion, tipoDeIdentificacion, sexo):
        from gestorAplicacion.general.Inventario import Inventario
        self._nombre = nombre
        self._telefono = telefono
        self._email = email
        self._identificacion = identificacion
        self._tipoDeIdentificacion = tipoDeIdentificacion
        self._sexo = sexo
        Inventario.agregarPersona(self)

    def getNombre(self):
        return self._nombre  
    def setNombre(self, nombre):
        self._nombre = nombre

    def getTelefono(self):
        return self._telefono  
    def setTelefono(self, telefono):
        self._telefono = telefono

    def getEmail(self):
        return self._email  
    def setEmail(self, email):
        self._email = email

    def getIdentificacion(self):
        return self._identificacion  
    def setIdentificacion(self, identificacion):
        self._identificacion = identificacion

    def getTipoDeIdentificacion(self):
        return self._tipoDeIdentificacion 
         
    def setTipoDeIdentificacion(self, tipoDeIdentificacion):
        self._nombre = tipoDeIdentificacion
    
    def getSexo(self):
        return self._sexo  
    def setSexo(self, sexo):
        self._sexo = sexo

    def mostrarInformacion(self):
        return "Soy " + self._nombre + "con número de identificación: " + self._identificacion

    def contratar(self, contrato: Contrato, cargo: str, servicio: Servicio, diasLaborales: list[DiaSemana]):
        from gestorAplicacion.general.Inventario import Inventario
        from gestorAplicacion.personas.Empleado import Empleado 
        Empleado(self._nombre, self._telefono, self._email, self._identificacion, self._tipoDeIdentificacion, self._sexo, contrato, cargo, servicio, diasLaborales)
        Inventario.eliminarPersona(self)
	